kratos/utilities/entities_utilities.h

```diff
@@ -32,6 +33,148 @@ namespace Kratos
+     * @tparam TEntity The entity type.
+     */
+    template<class TEntity>
+    class KRATOS_API(KRATOS_CORE) EntitityIdentifier
+    {
+    public:
+        ///@name Type Definitions
+        ///@{
+
+        /// Geometry type definition
+        using GeometryType = typename TEntity::GeometryType;
+
+        /// Pointer definition of ReplaceElementsAndConditionsProcess
+        KRATOS_CLASS_POINTER_DEFINITION(EntitityIdentifier);
+
+        ///@}
+        ///@name Life Cycle
+        ///@{
+            
+        /** 
+         * @brief Default constructor 
+         */
+        EntitityIdentifier() :
+            mName("")
+        {
+        }
+
+        /**
+         * @brief Constructor
+         * @param rName The name of the entity.
+         */
+        EntitityIdentifier(const std::string& rName);
+
+        /**
+         * @brief Copy constructor
+         */
+        EntitityIdentifier(const EntitityIdentifier& rOther)
+            : mpPrototypeEntity(rOther.mpPrototypeEntity),
+              mDefinitionType(rOther.mDefinitionType),
+              mName(rOther.mName),
+              mTypes(rOther.mTypes)
+        {
+        }
+
+        ///@}
+        ///@name Operators
+        ///@{
+
+        /**
+         * @brief Assignment operator.
+         */
+        EntitityIdentifier& operator=(const EntitityIdentifier& rOther)
+        {
+            mpPrototypeEntity = rOther.mpPrototypeEntity;
+            mDefinitionType = rOther.mDefinitionType;
+            mName = rOther.mName;
+            mTypes = rOther.mTypes;
+
+            return *this;
+        }
+
+        ///@}
+        ///@name Operations
+        ///@{
+
+        /**
+         * @brief Checks if the object is initialized.
+         * @return true if the object is initialized, false otherwise.
+         */
+        bool IsInitialized();
+
+        /**
+         * @brief Get the prototype entity.
+         * @param pGeometry The pointer to the geometry.
+         * @return const TEntity& The prototype entity.
+         */
+        const TEntity& GetPrototypeEntity(typename GeometryType::Pointer pGeometry);
+
+        ///@}
+        ///@name Input and output
+        ///@{
+
+        /// Turn back information as a string.
+        std::string Info() const
+        {
+            return "EntitityIdentifier";
+        }
+
+        /// Print information about this object.
+        void PrintInfo(std::ostream& rOStream) const
+        {
+            rOStream << "EntitityIdentifier";
+        }
+
+        /// Print object's data.
+        void PrintData(std::ostream& rOStream) const;
+
+        ///@}
+    private:
+        ///@name Member Variables
+        ///@{
+
+        typename TEntity::Pointer mpPrototypeEntity = nullptr;                    /// The prototype entity
+        DefinitionType mDefinitionType = DefinitionType::Single;                  /// The type of definition
```

Comment:
what is this??

Reply:
It is an enum, is defined some lines above